### PR TITLE
UX: Install dependencies, run servers, and build demos in a single command

### PR DESCRIPTION
**Pull request by yourslab**
To make for a better user experience, this configuration wizard allows the user to try all the demos by simply running this script rather than go through the endless number of steps in the README. To use:

```bash
./configure.sh
```

If you have the time, please try it out. You can run it from any directory :)

- Installs required dependencies and packages such as OpenSSL 1.1.x, CMake 3.x, and Docker while still being able to support multiple Operating Systems including Mac and non-Debian Linux Distros.
- Runs the MQTT and HTTP servers needed by calling `docker-compose` on `tools/local-servers/docker-compose.yml`
- Passes the configuration settings needed to connect to these servers through CMake flags.
- Saves these settings into a yaml file in case the user would like to use these settings again.

Tested on Ubuntu 16.04, Ubuntu 18.04, RHEL-8.2.0.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: demos/CMakeLists.txt [hunk 1]
```diff
@@ -19,20 +19,6 @@ function(check_aws_credentials demo_name)
...
-    unset(HAVE_CLIENT_CERT CACHE)
-    unset(HAVE_PRIVATE_KEY CACHE)
-    check_symbol_exists(AWS_IOT_ENDPOINT ${FILES_TO_CHECK} HAVE_AWS_ENDPOINT)
-    check_symbol_exists(CLIENT_CERT_PATH ${FILES_TO_CHECK} HAVE_CLIENT_CERT)
-    check_symbol_exists(CLIENT_PRIVATE_KEY_PATH ${FILES_TO_CHECK} HAVE_PRIVATE_KEY)
-    
-    if(NOT(HAVE_AWS_ENDPOINT AND HAVE_CLIENT_CERT AND HAVE_PRIVATE_KEY))
-        message("To run ${demo_name}, define AWS_IOT_ENDPOINT, CLIENT_CERT_PATH, CLIENT_PRIVATE_KEY_PATH in ${demo_name}/demo_config.h.")
-        set_target_properties(${demo_name} PROPERTIES EXCLUDE_FROM_ALL true)
-    endif()
```
Comment (yourslab) [inline demos/CMakeLists.txt:35]: Goodbye preprocessor logic :) @nateglims

File: demos/CMakeLists.txt [hunk 2]
```diff
@@ -19,20 +19,6 @@ function(check_aws_credentials demo_name)
...
     endif()
-
-    set(FILES_TO_CHECK "demo_config.h")
-    list(APPEND CMAKE_REQUIRED_INCLUDES "${CMAKE_CURRENT_LIST_DIR};${LOGGING_INCLUDE_DIRS}")
-    unset(HAVE_AWS_ENDPOINT CACHE)
-    unset(HAVE_CLIENT_CERT CACHE)
-    unset(HAVE_PRIVATE_KEY CACHE)
-    check_symbol_exists(AWS_IOT_ENDPOINT ${FILES_TO_CHECK} HAVE_AWS_ENDPOINT)
-    check_symbol_exists(CLIENT_CERT_PATH ${FILES_TO_CHECK} HAVE_CLIENT_CERT)
-    check_symbol_exists(CLIENT_PRIVATE_KEY_PATH ${FILES_TO_CHECK} HAVE_PRIVATE_KEY)
```
Comment (yourslab) [inline demos/CMakeLists.txt:30]: Goodbye preprocessor logic :) @nateglims

Comment (nateglims) [reply demos/CMakeLists.txt:30]: 👍

**Comment by sarenameas** [comment]
I like this can you include a README.md update?

**Comment by yourslab** [comment]
Closing as I feel that there is a better way to design this

**Pull request closed without merging**